client/app/components/sections/Topbar/Topbar.js

```diff
@@ -233,18 +263,31 @@ class Topbar extends Component {
       }) :
       {};
 
+    const oldSearchParams = parseSearchParams(location);
+
     return div({ className: css.topbar }, [
       this.props.menu ? r(MenuMobile, { ...mobileMenuProps, className: css.topbarMobileMenu }) : null,
       r(Logo, { ...this.props.logo, classSet: css.topbarLogo, color: marketplace_color1 }),
       div({ className: css.topbarMediumSpacer }),
-      SEARCH_ENABLED && this.props.search ?
+      this.props.search ?
         r(SearchBar, {
           mode: this.props.search.mode,
-          keywordPlaceholder: this.props.search.keyword_placeholder,
-          locationPlaceholder: this.props.search.location_placeholder,
-          onSubmit: this.props.search.onSubmit || (() => {
-            console.log('submit search'); // eslint-disable-line no-console
-          }),
+          keywordPlaceholder: t('web.topbar.search_placeholder'),
+          locationPlaceholder: t('web.topbar.search_location_placeholder'),
+          keywordQuery: oldSearchParams.q,
+          locationQuery: oldSearchParams.lq,
+          customColor: marketplace_color1,
+          onSubmit: ({ keywordQuery, locationQuery, place }) => {
+            const query = createQuery({
+              q: keywordQuery,
+              lq: locationQuery,
+              lc: placesUtils.coordinates(place),
+              boundingbox: placesUtils.viewport(place),
+              distance_max: placesUtils.maxDistance(place),
```

Comment:
One additional parameter to consider is service status 
(we used that to figure out Google maps related API problems like quota etc.)
However, it can be added after dl if needed.